charts/ai-software-templates/chatbot/templates/application-gitops-job.yaml

```diff
@@ -12,7 +12,85 @@ spec:
     spec:
       containers:
         - name: repo-creator
-          image: quay.io/redhat-ai-dev/helm-chart-application-gitops:latest
+          command:
+            - /bin/bash
+            - -c
+            - |
+              #!/usr/bin/env bash
+              set -o nounset
+              set -o pipefail
+              cd /tmp || exit
+              wget https://github.com/cli/cli/releases/download/v2.62.0/gh_2.62.0_linux_amd64.tar.gz || exit
+              tar xzf gh_2.62.0_linux_amd64.tar.gz
+              export PATH=$(pwd)/gh_2.62.0_linux_amd64/bin:$PATH
+              export SOURCE_REPO_APP_CONTENT_PATH="chatbot"
+              export SOURCE_TEKTON_REPO_CONTENT_PATH="pac/pipelineRuns"
+              export TEKTON_FILE_APP_NAME_REPLACEMENT="application_name_replace"
+              export TEKTON_FILE_APP_NAMESPACE_REPLACEMENT="application_namespace_replace"
+              export TEKTON_FILE_QUAY_ACCOUNT_REPLACEMENT="quay_account_replace"
+              echo APP_NAME is "$APP_NAME"
+              echo APP_NAMESPACE is "$APP_NAMESPACE"
+              echo GITHUB_ORG_NAME is "$GITHUB_ORG_NAME"
+              echo GITHUB_SOURCE_REPO is "$GITHUB_SOURCE_REPO"
+              echo GITHUB_TETKON_SOURCE_REPO is "$GITHUB_TETKON_SOURCE_REPO"
+              echo GITHUB_DEFAULT_BRANCH is "$GITHUB_DEFAULT_BRANCH"
+              echo QUAY_ACCOUNT_NAME is "$QUAY_ACCOUNT_NAME"
+
+              export GH_TOKEN=$GITHUB_TOKEN
+              export GH_REPO="github.com/$GITHUB_ORG_NAME/$APP_NAME"
+
+              export HOME=/tmp
+
+              gh auth login -p https
+              gh auth setup-git --hostname=github.com
+
+              gh repo view "$GH_REPO" --json id 2> /dev/null
+
+              return_code=$?
+
+              if [ $return_code == 0 ]; then
+                echo Repo "$GH_REPO" already exists
+                exit 0
```

Comment:
I think if repo already exists we should return with non 0 🤔

Reply:
I was admittedly on the fence here

Perhaps the better solution would be to checkout the repo and see if it has the contents we would have put in there, vs. the user giving a name that was already picked.

Maybe see what others think before making a change ?

Reply:
I had the same thought tbh when I was creating the python image too (return 0 or return 1). I chose 1 to be more aligned with the software template, as the initial version of the helm chart is a conversion of the software template.

> Perhaps the better solution would be to checkout the repo and see if it has the contents we would have put in there, vs. the user giving a name that was already picked.

I think it's a good idea, especially if for any reason their helm release was accidentically deleted or anything they can just re-deploy using the same repo (after all we wiil check that all the contents are there).

Reply:
I'll see if I can sort out the content inspection and if so add that.

Otherwise sure I'll return 1 if the repo exists / not up to snuff, etc.